tree/30_generic_methods/logger_rudder.cf

```diff
@@ -50,9 +51,25 @@ bundle agent logger_rudder(message, class_prefix)
         int        => getfields("^${current_technique_report_info.technique_name};;${c_class_prefix};;.*", "${expected_reports_file}", ";;", "report_data");
 
   classes:
-      "report_data_read" expression => isgreaterthan("${number_lines}","0");
+      "report_data_read"    expression => isgreaterthan("${number_lines}","0");
+  
+      # method promiser has not been rewritten
+      "method_v2"                  not => strcmp("method_call", nth("this.callers_promisers", "1"));
 
-      "keys_defined"     expression => isvariable("keys");
+      # True if the csv file contains at least 6 fields (or none that match)
+      # The res file will contain the unexpanded reports variable if v2 has not been generated
+      # 5 vs 6 : because getfields and readstringarrayidx do not have the same indexing method (getfields starts at 1)
```

Comment:
Ok.
More in detail, the file content in created below, so a missing 6th column will translate to the litteral value "${reports[0][5]}" (and not 6 because of discrepancy in indexing)